src/compiler/compile/render_dom/wrappers/Element/index.ts

```diff
@@ -906,6 +907,29 @@ export default class ElementWrapper extends Wrapper {
 		});
 	}
 
+	add_styles(block: Block) {
+		this.node.styles.forEach((style_directive) => {
+			const { name, expression, text } = style_directive;
+
+			let snippet;
+			if (text) {
+				snippet = `"${text}"`;
+			} else if (expression) {
+				snippet = expression.manipulate(block);
+			} else {
+				snippet = name;
+			}
+
+			const updater = b`@set_style(${this.var}, "${name}", ${snippet}, false)`;
+
+			block.chunks.hydrate.push(updater);
+
+			if (expression && expression.dynamic_dependencies().length) {
+				block.chunks.update.push(updater);
```

Comment:
hmm, not sure how I would do this...

Reply:
1. you can check https://github.com/sveltejs/svelte/blob/b6a74a0ac5b1798c49eafda2263a64025da709ab/src/compiler/compile/render_dom/wrappers/Element/index.ts#L879-L881

where, you should apply the style changes regardless if there's a spread. if there's a `style` attribute, it will be applied part of the spread operation, so you need to re-apply the `updater` to override that.

2. to check if the attribute value has changed before calling `@set_style`, you can learn from here:

https://github.com/sveltejs/svelte/blob/b6a74a0ac5b1798c49eafda2263a64025da709ab/src/compiler/compile/render_dom/wrappers/Element/index.ts#L879-L881

calling `block.renderer.dirty(all_dependencies)` gives you an expression that checks if the dependencies has changed, and you use it like this:

```js
block.chunks.update.push(b`
  if (${condition}) {
	  ${updater}
  }
`);
```

Reply:
i've made the relevant changes over here: 

https://github.com/sveltejs/svelte/blob/c3b6140cce00017b33f6af7dca795e1f150d2eca/src/compiler/compile/render_dom/wrappers/Element/index.ts#L937-L941